### PR TITLE
Expose named groups from Pattern

### PR DESCRIPTION
**Pull request by j4ckcyw**
https://github.com/google/re2j/issues/125

### CODE REVIEW

File: java/com/google/re2j/Pattern.java [hunk 1]
```diff
@@ -269,6 +270,14 @@ public int groupCount() {
     return re2.numberOfCapturingGroups();
   }
 
+  /**
+   * Return a map of the capturing groups in this matcher's pattern, where key is the name and value
+   * is the index of the group in the pattern.
+   */
+  public Map<String, Integer> namedGroups() {
+    return re2.namedGroups;
```
Comment (sjamesr) [inline java/com/google/re2j/Pattern.java:278]: should we defensively copy the map to ensure that callers don't change it?

Comment (j4ckcyw) [reply java/com/google/re2j/Pattern.java:278]: done.

**Comment by sjamesr** [comment]
Thank you!

**Comment by j4ckcyw** [comment]
Fixes #125

**Pull request merged**